src/state/currentTask.ts

```diff
@@ -1,374 +1,200 @@
-                // findActiveTab give a new reference to activeTab every time
-                activeTab = await findActiveTab();
-                return activeTab?.status === "complete";
-              },
-              200, // check every 200ms
-              100, // wait for up to 20 seconds (100*200ms)
-              false, // assume page fully loaded on timeout
-            );
-          }
-
-          const performAction = async (
-            query: QueryResult,
-          ): Promise<boolean> => {
-            if (query == null) {
-              set((state) => {
-                state.currentTask.status = "error";
-              });
-              return false;
-            }
-
-            setActionStatus("performing-action");
-            if (voiceMode) {
-              if ("speak" in query.action && query.action.speak) {
-                voiceControl.speak(query.action.speak, onError);
-              } else if ("thought" in query.action && query.action.thought) {
-                voiceControl.speak(query.action.thought, onError);
-              }
-            }
-
-            set((state) => {
-              query &&
-                state.currentTask.history.push({
-                  prompt: query.prompt,
-                  response: query.rawResponse,
-                  action: query.action,
-                  usage: query.usage,
-                });
-            });
-            if (
-              query.action.operation === null ||
-              query.action.operation.name === "finish" ||
-              query.action.operation.name === "fail"
-            ) {
-              return false;
-            }
-            if (agentMode === AgentMode.VisionEnhanced) {
-              await operateTool(tabId, query.action.operation);
-            } else {
-              await operateToolWithSimpliedDom(tabId, query.action.operation);
-            }
-            return true;
-          };
-
-          let query: QueryResult | null = null;
-
-          // check if the tab does not allow attaching debugger, e.g. chrome:// pages
-          if (activeTab.url?.startsWith("chrome")) {
-            setActionStatus("generating-action");
-            query = await determineNavigateAction(instructions);
-
-            if (wasStopped()) break;
-
-            const shouldContinue = await performAction(query);
-            if (shouldContinue) {
-              // if navigation was successful, continue the task on the new page
-              setActionStatus("waiting");
-              continue;
-            } else {
-              break;
-            }
-          }
-          await attachDebugger(tabId);
-          await waitTillHTMLRendered(tabId);
-
-          set((state) => {
-            state.currentTask.tabId = tabId;
-          });
-
-          const previousActions = get()
-            .currentTask.history.map((entry) => entry.action)
-            .filter(truthyFilter);
-
-          if (agentMode === AgentMode.VisionEnhanced) {
```

Comment:
You've completed forgotten logics related to `agentMode` when refactoring. Please fix it.